packages/block-library/src/navigation-link/edit.js

```diff
@@ -518,11 +517,6 @@ export default function NavigationLinkEdit( {
 												'core/image',
 												'core/strikethrough',
 											] }
-											onClick={ () => {
-												if ( ! url ) {
-													setIsLinkOpen( true );
-												}
-											} }
```

Comment:
There is a ! url check on line 478 that makes this section never render if there's no url, so this check will never be valid.